dts/arm/silabs/efr32mg21.dtsi

```diff
@@ -41,9 +41,9 @@
 	};
 
 	soc {
-		msc: flash-controller@40030000 {
```

Comment:
can you explain why a HAL update affects register addresses?

Reply:
Addresses starting from `0x40004000` to `0x4C022000` are handled by Security Management Unit as secured memory regions while addresses from `0x50004000` to `0x5C022000` are non-secure.
After HAL update, register addresses had to be changed because the defines for the base addresses in HAL were renamed.
Some of the existing drivers use HAL functions that rely on defines from HAL. Previously, those defines were pointing to secured address space but after the rename those are in non-secured space by default. That caused conflicts with reg addresses in device tree and e.g. UART driver stopped working. I had to modify DTS to fix this.